sync-core/src/main/java/com/cloudant/mazha/HttpRequests.java

```diff
@@ -110,11 +112,29 @@ public int getMaxForRoute(HttpRoute route) {
         addDebuggingInterceptor(httpClient);
         this.httpClient = httpClient;
 
-        if (!Strings.isNullOrEmpty(username)  &&  !Strings.isNullOrEmpty(password)) {
+        String authHeaderValue = null;
+        if (!Strings.isNullOrEmpty(username) && !Strings.isNullOrEmpty(password)) {
             String authString = username + ":" + password;
             Base64 base64 = new Base64();
             authHeaderValue = "Basic " + new String(base64.encode(authString.getBytes()));
         }
+
+        if (customHeaders != null) {
+            this.requestHeaders = customHeaders;
+            if (customHeaders.containsKey("authorization")) {
+                if (authHeaderValue != null) {
+                    // user supplied a custom auth header as well as user/pass - custom auth header takes precedence
+                    logger.warning("custom header \"authorization\" and username/password are both set; will use the value from custom header");
+                }
+            } else {
+                if (authHeaderValue != null) {
+                    // if we're doing basic auth, make a copy of the custom headers and add it in
+                    this.requestHeaders = new HashMap<String, String>();
+                    customHeaders.putAll(customHeaders);
+                    customHeaders.put("authorization", authHeaderValue);
+                }
+            }
+        }
```

Comment:
It looks like here the Authorization header will not be set if there are no already-existing custom headers (`customHeaders != null`).